### PR TITLE
Modified SensorTag example and added some examples derived from 1.0.7

### PR DESCRIPTION
**Pull request by noelpaz**
@SidLeung Please review

### CODE REVIEW

**Comment by noelpaz** [comment]
Internal JSON is 155

File: libraries/CurieBLE/examples/central/BatteryMonitor_Central/BatteryMonitor_Central.ino [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+void processNotification(BLEDevice peripheral) {
+  if (peripheral.connect()) {
+    Serial.print("Connected");
+    Serial.println(peripheral.address());
+    // light pin to indicate connection
+    digitalWrite(LED_PIN, HIGH);
+  } else {
+    Serial.println("Failed to connect!");
+    return;
```
Comment (SidLeung) [inline libraries/CurieBLE/examples/central/BatteryMonitor_Central/BatteryMonitor_Central.ino]: Do you want to turn off LED here?

File: libraries/CurieBLE/examples/central/BatteryMonitor_Central/BatteryMonitor_Central.ino [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  if (!batteryLevelChar) {
+    peripheral.disconnect();
+    Serial.println("Peripheral does not have battery level characteristic!");
+    digitalWrite(LED_PIN, LOW);
+    delay(1000);
+    return;
+  }
+
+  batteryLevelChar.subscribe();
+
```
Comment (SidLeung) [inline libraries/CurieBLE/examples/central/BatteryMonitor_Central/BatteryMonitor_Central.ino]: Check for return value, may be?

File: libraries/CurieBLE/examples/central/sensortag_button/sensortag_button.ino [hunk 3]
```diff
@@ -45,8 +64,20 @@ void loop() {
...
+      In this release we do not have the feature that gets the scan response data and hence
+      the local name in the scan is blank
+      We have to explicitly find the BLE mac address
+      Please use another deviice like nrfConnect app to discover the Bluetooth Address
+    */
+    //if (peripheral.localName() == "SensorTag") {
+
+
+    // change to the mac address according to your device
+    if (peripheral.address() == "24:71:89:07:27:80")
```
Comment (SidLeung) [inline libraries/CurieBLE/examples/central/sensortag_button/sensortag_button.ino]: Do you think we need a big sign here asking the user to change the MAC addr?

**Comment by SidLeung** [comment]
@noelpaz Looks good.  Really minor stuffs.  It's up to you to correct them.

File: libraries/CurieBLE/examples/central/BatteryMonitor_Central/BatteryMonitor_Central.ino [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  {
+    Serial.print("Found ");
+    Serial.print(peripheral.address());
+    Serial.print(" '");
+    Serial.print(peripheral.localName());
+    Serial.print("' ");
+    Serial.print(peripheral.advertisedServiceUuid());
+    Serial.println();
+    if ( peripheral.localName() == "BatteryMonitorSketch") {
+      BLE.stopScan();
```
Comment (noelpaz) [inline libraries/CurieBLE/examples/central/BatteryMonitor_Central/BatteryMonitor_Central.ino]: For adopted UUID it actually does

File: libraries/CurieBLE/examples/central/BatteryMonitor_Central/BatteryMonitor_Central.ino (hunk 2 above)
Comment (noelpaz) [inline libraries/CurieBLE/examples/central/BatteryMonitor_Central/BatteryMonitor_Central.ino]: Will fix

File: libraries/CurieBLE/examples/central/sensortag_button/sensortag_button.ino [hunk 5]
```diff
@@ -19,13 +19,32 @@
 
 #include <CurieBLE.h>
 
+#define NUM_OF_SERVICE    10
+char *serviceUUIDArray[NUM_OF_SERVICE] =
```
Comment (noelpaz) [inline libraries/CurieBLE/examples/central/sensortag_button/sensortag_button.ino]: Will fix

File: libraries/CurieBLE/examples/central/sensortag_button/sensortag_button.ino (hunk 3 above)
Comment (noelpaz) [inline libraries/CurieBLE/examples/central/sensortag_button/sensortag_button.ino]: Will fix Box comment

**Pull request merged**